src/App.tsx

```diff
@@ -1,54 +1,135 @@
+    }, [],
+  );
+
+  const handleSelect = useCallback(
+    (
+      event: React.MouseEvent<HTMLAnchorElement>,
+      person: Person,
+    ) => {
+      event.preventDefault();
+      setSelectedPerson(person);
+      setQuery(person.name);
+      setIsDropdownVisible(false);
+    }, [],
+  );
+
+  const handleReset = () => {
+    setQuery('');
+    setSelectedPerson(null);
+  };
+
+  useEffect(() => {
+    const timerId = setTimeout(() => {
+      setIsDropdownVisible(!query || isInputFocused);
+    }, 500);
+
+    return () => clearTimeout(timerId);
+  }, [query, isInputFocused, handleFocus, handleBlur]);
 
   return (
     <main className="section">
       <h1 className="title">
-        {`${name} (${born} = ${died})`}
+        {selectedPerson
+          ? `${selectedPerson.name} (${selectedPerson.born} - ${selectedPerson.died})`
+          : 'No selected person'}
       </h1>
 
-      <div className="dropdown is-active">
-        <div className="dropdown-trigger">
+      <div className={`dropdown ${isDropdownVisible ? 'is-active' : ''}`}>
+        <div className="dropdown-trigger has-icons-right">
           <input
             type="text"
             placeholder="Enter a part of the name"
             className="input"
+            value={query}
+            onChange={handleInput}
+            onBlur={handleBlur}
+            onFocus={handleFocus}
           />
+          {!!query && (
+            <span className="icon is right">
+              <button
+                className="delete is small"
+                type="button"
+                onClick={handleReset}
+              >
+                x
+              </button>
+            </span>
+          )}
         </div>
 
         <div className="dropdown-menu" role="menu">
           <div className="dropdown-content">
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter Bernard Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter Antone Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-danger">Elisabeth Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter de Decker</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-danger">Petronella de Decker</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-danger">Elisabeth Hercke</p>
-            </div>
+            {filteredPeople.length === 0 && (
+              <div className="dropdown-item">
+                <p>No people found.</p>
+              </div>
+            )}
+            {filteredPeople.map(person => (
```

Comment:
don't show the dropdown before you type something 
( I mean when I open the demo link your dropdown is already open but it should be closed)

use `isDropdownVisible` condition for that